pkg/me/me.go

```diff
@@ -0,0 +1,273 @@
+package me
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/9elements/converged-security-suite/v2/pkg/ostools"
+)
+
+type LegacyFlashPartitionTableHeader struct {
+	Padding        [16]uint8 // 16 zeros
+	Marker         uint32    // Always $FPT
+	NumFptEntries  uint32
+	HeaderVersion  uint8
+	EntryVersion   uint8
+	HeaderLength   uint8 // Usually 0x30
+	HeaderChecksum uint8
+	TicksToAdd     uint16
+	TokensToAdd    uint16
+	UMASize        uint32
+	Flags          uint32
+}
+
+func (h LegacyFlashPartitionTableHeader) String() string {
+
+	ret := "Flash partition table:\n"
+	ret += fmt.Sprintf(" Entries: %d\n", h.NumFptEntries)
+	ret += fmt.Sprintf(" HeaderVersion: 0x%x\n", h.HeaderVersion)
+	ret += fmt.Sprintf(" EntryVersion: 0x%x\n", h.EntryVersion)
+	ret += fmt.Sprintf(" HeaderLength: 0x%x\n", h.HeaderLength)
+	ret += fmt.Sprintf(" HeaderChecksum: 0x%x\n", h.HeaderChecksum)
+	ret += fmt.Sprintf(" TicksToAdd: 0x%x\n", h.TicksToAdd)
+	ret += fmt.Sprintf(" TokensToAdd: 0x%x\n", h.TokensToAdd)
+	ret += fmt.Sprintf(" UMASize: 0x%x\n", h.UMASize)
+	ret += fmt.Sprintf(" Flags: 0x%x\n", h.Flags)
+
+	return ret
+}
```

Comment:
Please use strings.Builder with WriteString and/or fmt.Fprintf

Reply:
done